nbconvert/templates/markdown.tpl

```diff
@@ -8,7 +8,9 @@
 {%- endblock output_prompt %}
 
 {% block input %}
-{{ cell.source | indent(4)}}
+```{% if nb.metadata.kernelspec %}{{ nb.metadata.kernelspec.language }}{% endif %}
```

Comment:
I think we should be using `metadata.language_info.name` instead of `metadata.kernelspec.language`. The latter is really just an ID to find a sensible substitute kernel when we can't find one matching `kernelspec.name`.